packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -133,6 +133,8 @@ class MessagePayload {
       }
     }
 
+    const enforce_nonce = Boolean(this.options.enforceNonce);
```

Comment:
Perhaps we should throw an error if `enforceNonce` is `true` but no `nonce` was passed?